build_script.py

```diff
@@ -36,146 +37,218 @@ def _mkdirp(path):
-    given 'swiftc' compiler.
-    """
-    lit_path = os.path.abspath(args.lit)
-    if not os.path.exists(lit_path):
-        raise IOError(
-            'Could not find lit tester tool at path: "{}". This tool is '
-            'requred to run the test suite. Unless you specified a custom '
-            'path to the tool using the "--lit" option, the lit tool will be '
-            'found in the LLVM source tree, which is expected to be checked '
-            'out in the same directory as swift-corelibs-xctest. If you do '
-            'not have LLVM checked out at this path, you may follow the '
-            'instructions for "Getting Sources for Swift and Related '
-            'Projects" from the Swift project README in order to fix this '
-            'error.'.format(lit_path))
-
-    # FIXME: Allow these to be specified by the Swift build script.
-    lit_flags = "-sv --no-progress-bar"
-    tests_path = os.path.join(SOURCE_DIR, "Tests", "Functional")
-    core_foundation_build_dir = _core_foundation_build_dir(
-        args.foundation_build_dir)
-
-    run('SWIFT_EXEC={swiftc} '
-        'BUILT_PRODUCTS_DIR={built_products_dir} '
-        'FOUNDATION_BUILT_PRODUCTS_DIR={foundation_build_dir} '
-        'CORE_FOUNDATION_BUILT_PRODUCTS_DIR={core_foundation_build_dir} '
-        '{lit_path} {lit_flags} '
-        '{tests_path}'.format(
-            swiftc=args.swiftc,
-            built_products_dir=args.build_dir,
-            foundation_build_dir=args.foundation_build_dir,
-            core_foundation_build_dir=core_foundation_build_dir,
-            lit_path=lit_path,
-            lit_flags=lit_flags,
-            tests_path=tests_path))
-
-
-def _install(args):
-    """
-    Install the XCTest.so, XCTest.swiftmodule, and XCTest.swiftdoc build
-    products into the given module and library paths.
-    """
-    build_dir = os.path.abspath(args.build_dir)
-    module_install_path = os.path.abspath(args.module_install_path)
-    library_install_path = os.path.abspath(args.library_install_path)
-
-    _mkdirp(module_install_path)
-    _mkdirp(library_install_path)
-
-    xctest_so = "libXCTest.so"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_so),
-        os.path.join(library_install_path, xctest_so)))
-
-    xctest_swiftmodule = "XCTest.swiftmodule"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_swiftmodule),
-        os.path.join(module_install_path, xctest_swiftmodule)))
-
-    xctest_swiftdoc = "XCTest.swiftdoc"
-    run("cp {} {}".format(
-        os.path.join(build_dir, xctest_swiftdoc),
-        os.path.join(module_install_path, xctest_swiftdoc)))
+class DarwinStrategy:
+    @staticmethod
+    def requires_foundation_build_dir():
+        """
+        The Foundation build directory is not required on Darwin because the
+        Xcode workspace implicitly builds Foundation when building the XCTest
+        schemes.
+        """
```

Comment:
nit-pick: This is a great explanation, but it seems like more a `# source comment` than `"""documentation"""`.